### DISCUSSION TITLE
alternate-pin

### DISCUSSION THREAD

**Discussion by robzed** [tabemann/zeptoforth#69] [General]
I don't really get what alternate pin is for?

```
  \ Set an alternate function
  : alternate-pin ( function pin -- )
    dup validate-pin
    over validate-alternate
    true over PADS_BANK0_IE!
    false over PADS_BANK0_OD!
    GPIO_CTRL_FUNCSEL!
  ;
```

Surely if you want an alternative function, i.e. not an input pin or an output pin, surely you don't want Input enabled (IE) and Output enabled (OD)?

I'd expect:
```
    false over PADS_BANK0_IE!
    true over PADS_BANK0_OD!
```

What does everyone else think?

**Comment by tabemann** [comment]
I do not remember where I read that for alternate functions Input Enable should be true and Output Disable should be false (it is probably buried somehere in the reference manual where I am too lazy to find it), but just from searching the Interwebs I got [this](https://forums.raspberrypi.com/viewtopic.php?t=312533#p1869300) which indicates that this should be so.

**Reply by robzed** [reply]
There is a lot of clever things on this chip .... like the ADC bits are nibble aligned so they are easy to see in Hex. 

But there are also some thing you'd never guess!

Been working with the ADC FIFO which the RP2040 data sheet says it's 4 levels. But I'm getting 8....  with separate readings (and the levels says 8)....
